src/java.base/share/classes/sun/security/ssl/SignatureScheme.java

```diff
@@ -375,10 +387,32 @@ static List<SignatureScheme> getSupportedAlgorithms(
             List<ProtocolVersion> activeProtocols) {
         List<SignatureScheme> supported = new LinkedList<>();
 
-        List<SignatureScheme> schemesToCheck =
-                config.signatureSchemes == null ?
-                    Arrays.asList(SignatureScheme.values()) :
-                    namesOfAvailable(config.signatureSchemes);
+        // If config.signatureSchemes is non-empty then it means that
+        // it was defined by a System property.  Per
+        // SSLConfiguration.getCustomizedSignatureScheme() the list will
+        // only contain schemes that are in the enum.
+        // Otherwise, use the enum constants (converted to a List).
+        //
+        // Additional logic is added here to remove the ecdsa_brainpoolP512r1tls13_sha512
+        // signature scheme by default. We only want to make use of ecdsa_brainpoolP512r1tls13_sha512
+        // when explicitly set via System property.
+        List<SignatureScheme> schemesToCheck = null;
+        if (config.signatureSchemes == null) {
+            List<SignatureScheme> schemes = Arrays.asList(SignatureScheme.values());
+            schemesToCheck = new ArrayList<>(schemes.size());
+            for (SignatureScheme scheme: schemes) {
+                if (scheme != ECDSA_BRAINPOOLP512R1TLS13_SHA512) {
+                    schemesToCheck.add(scheme);
+                } else {
+                    if (SSLLogger.isOn &&
+                        SSLLogger.isOn("ssl,handshake,verbose")) {
+                        SSLLogger.finest("Ignore " + ECDSA_BRAINPOOLP512R1TLS13_SHA512.name + " from supported signature schemes");
+                    }
+                }
+            }
+        } else {
+            schemesToCheck = namesOfAvailable(config.signatureSchemes);
+        }
```

Comment:
Suggest inverting this `if` statement so both branches are close to the condition.
Also:
* the initializer for `schemesToCheck` is redundant
* we should have a space both sides of `:`
* line 409 should be folded
```java
        List<SignatureScheme> schemesToCheck;
        if (config.signatureSchemes != null) {
            schemesToCheck = namesOfAvailable(config.signatureSchemes);
        } else {
            List<SignatureScheme> schemes = Arrays.asList(SignatureScheme.values());
            schemesToCheck = new ArrayList<>(schemes.size());
            for (SignatureScheme scheme : schemes) {
                if (scheme != ECDSA_BRAINPOOLP512R1TLS13_SHA512) {
                    schemesToCheck.add(scheme);
                } else {
                    if (SSLLogger.isOn &&
                        SSLLogger.isOn("ssl,handshake,verbose")) {
                        SSLLogger.finest("Ignore " + ECDSA_BRAINPOOLP512R1TLS13_SHA512.name
                                + " from supported signature schemes");
                    }
                }
            }
        }
```

Reply:
Inverted the logic here as suggested along with folding long line suggested.